### PR TITLE
Implement lint/go command for local

### PR DESCRIPTION
**Pull request by knanao**
**What this PR does / why we need it**:
In fact, we have to fix all lint errors before using this command.
So let me fix all of them one by one in another PR.
https://github.com/pipe-cd/pipecd/runs/10154286748

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -138,8 +138,15 @@ run/site:
 # Lint commands
 
 .PHONY: lint/go
+lint/go: FIX ?= false
+lint/go: VERSION ?= sha256:78d1bbd01a9886a395dc8374218a6c0b7b67694e725dd76f0c8ac1de411b85e8 #v1.46.2
+lint/go: FLAGS ?= --rm -e GOLANGCI_LINT_CACHE=/repo/.cache/golangci-lint -v ${PWD}:/repo -w /repo -it
 lint/go:
-	@echo "Unimplemented"
+ifeq ($(FIX),true)
+	docker run ${FLAGS} golangci/golangci-lint@${VERSION} golangci-lint run --fix
```
Comment (khanhtc1202) [inline Makefile:146]: Should we add `--rm` to remove the container after the job is done? 🤔

Comment (knanao) [reply Makefile:146]: this one?
https://github.com/pipe-cd/pipecd/pull/4086/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R143

Comment (khanhtc1202) [reply Makefile:146]: Ah true, I missed reading that part 👍

**Review by khanhtc1202 — APPROVED** [review]
Nice improvement 👍

**Pull request merged**